dropshot/tests/test_multipart.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 Oxide Computer Company
+
+//! Test cases for multipart form-data.
+
+use dropshot::test_util::read_string;
+use dropshot::{
+    endpoint, ApiDescription, HttpError, MultipartBody, RequestContext,
+};
+use http::{Method, Response, StatusCode};
+use hyper::Body;
+
+extern crate slog;
+
+pub mod common;
+
+fn api() -> ApiDescription<usize> {
+    let mut api = ApiDescription::new();
+    api.register(api_multipart).unwrap();
+    api
+}
+
+#[endpoint {
+    method = POST,
+    path = "/upload",
+}]
+async fn api_multipart(
+    _rqctx: RequestContext<usize>,
+    mut body: MultipartBody,
+) -> Result<Response<Body>, HttpError> {
+    // Iterate over the fields, use `next_field()` to get the next field.
+    if let Some(field) = body.content.next_field().await.unwrap() {
+        // Process the field data chunks e.g. store them in a file.
+        if let Ok(bytes) = field.bytes().await {
+            return Ok(Response::builder()
```

Comment:
Now that I think about it, maybe this test should concatenate all the fields together, so that we know it doesn't crash after the first field. I think that would increase test coverage.